qemu/tests/numa_prealloc_threads.py

```diff
@@ -46,24 +62,36 @@ def run(test, params, env):
     expected_cpu_affinity = thread_context_device.get_param("cpu-affinity")
     cpu_affinity = vm.monitor.qom_get(thread_context_device_id, "cpu-affinity")
     affinity = str(cpu_affinity[0]) + "-" + str(cpu_affinity[-1])
-    test.log.debug("The affinity: %s and the expected_cpu_affinity: %s"
-                   % (affinity, expected_cpu_affinity))
+    error_context.context("The affinity: %s and the expected_cpu_affinity: %s"
+                          % (affinity, expected_cpu_affinity), test.log.debug)
     if expected_cpu_affinity != affinity:
         test.fail("Test and QEMU cpu-affinity does not match!")
 
-    cmd_taskset = "taskset -c -p " + str(thread_id)
-    output = process.getoutput(cmd_taskset)
-    if not re.search(affinity, output):
-        test.fail("The affinities %s and %s do not match!"
-                  % (affinity, str(output)))
+    cmd_taskset = "taskset -c -p %s" % thread_id
+    check_affinity(affinity, cmd_taskset, test)
 
     sandbox = params.get("qemu_sandbox", "on")
     if sandbox == "on":
         try:
             # The command is expected to fail
-            vm.monitor.qom_set(thread_context_device_id, "cpu-affinity", cpu_affinity)
+            vm.monitor.qom_set(thread_context_device_id,
+                               "cpu-affinity",
+                               cpu_affinity)
         except QMPCmdError as e:
-            test.log.debug("The expected error message: %s and the output: %s"
-                           % (error_msg, e.data))
+            error_context.context("The expected error message: %s and the output: %s"
+                                  % (error_msg, e.data), test.log.debug)
             if not re.search(error_msg, str(e.data)):
-                test.fail("Can not get expected error message: %s" % error_msg)
+                test.fail("Cannot get expected error message: %s" % error_msg)
+
+    check_affinity(affinity, cmd_taskset, test)
+    error_context.context("Set externally a new CPU affinity", test.log.info)
+    cmd_taskset = "taskset -c -p %s %s" % (new_cpu_affinity, str(thread_id))
+    output = process.getoutput(cmd_taskset)
+    if not re.search("new affinity list: %s" % new_cpu_affinity, output):
+        test.fail("The new affinity %s is not correct!" % new_cpu_affinity)
+
+    cmd_taskset = "taskset -c -p $(pgrep qemu)"
+    output = process.getoutput(cmd_taskset)
+    # QEMU main thread value should prevail as 0
+    if not re.search("current affinity list: %d" % 0, output):
```

Comment:
Simplified to "current affinity list: 0"?

Reply:
Updated, thanks !